### PR TITLE
CheckTestsOnFixedAppVersion

### PR DESCRIPTION
**Pull request by Kristinochka-bee**

### CODE REVIEW

File: src/test/java/pages/AddNewCoursePage.java [hunk 1]
```diff
@@ -20,12 +25,16 @@ public class AddNewCoursePage extends HomePage{
...
+
+    public void addNewCourse(String courseName, String faculty, String courseDescription, String courseStart, String courseEnd){
         addNewCourseTable.shouldBe(Condition.visible);
         courseNameInput.sendKeys(courseName);
-        selectFaculty(faculty);
+        facultyDropdown.click();
+        facultyList.findBy(Condition.exactText(faculty)).click();
         courseDescriptionField.sendKeys(courseDescription);
-        coverPhotoForm.sendKeys("images/qa..jpg");
+        coverPhotoForm.sendKeys("C:\\work\\GraduationProject\\SELENIUM_SELENID\\src\\test\\java\\images\\qa..jpg");
```
Comment (LeonadO555) [inline src/test/java/pages/AddNewCoursePage.java]: will be better transfer to variable 
"C:\\work\\GraduationProject\\SELENIUM_SELENID\\src\\test\\java\\images\\qa..jpg"

File: src/test/java/pages/CourseDeatailsPage.java [hunk 2]
```diff
@@ -10,21 +11,39 @@
...
+    }
+
+    public void clickOnSelectedDocument(String link){
+        $x("//*[@id=\"course-material-list\"]//div[contains(text(),"+ link +")]").click();
+    }
 
     public void uploadCourseMaterial(String courseName){
         documentNameInput.sendKeys(courseName);
-        uploadDocumentField.sendKeys("files/qa.txt");
+        uploadDocumentField.sendKeys("C:\\work\\GraduationProject\\SELENIUM_SELENID\\src\\test\\java\\files\\qa.txt");
```
Comment (LeonadO555) [inline src/test/java/pages/CourseDeatailsPage.java]: will be better transfer to variable
"C:\\work\\GraduationProject\\SELENIUM_SELENID\\src\\test\\java\\files\\qa.txt"

File: src/test/java/tests/e2e/TeacherCanUploadCourseMaterialAndStudentCanDownLoadIt.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        homePage.coursesDropdownMenu.click();
+        homePage.courseListButton.shouldBe(Condition.visible);
+        homePage.courseListButton.click();
+        courseListPage = new CourseListPage();
+        courseListPage.searchBox.shouldBe(Condition.visible);
+        courseListPage.searchCourseByName(facultyName);
+        courseListPage.getSearchedFacultyCourseInTable().shouldBe(Condition.visible);
+        courseListPage.courseCard.click();
+        courseDeatailsPage = new CourseDeatailsPage();
+        Thread.sleep(1000);
```
Comment (LeonadO555) [inline src/test/java/tests/e2e/TeacherCanUploadCourseMaterialAndStudentCanDownLoadIt.java]: need to replace for waiting for any element

File: src/test/java/tests/e2e/TeacherCanUploadCourseMaterialAndStudentCanDownLoadIt.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    }
+
+    @Test
+    public void studentCanViewDocumentsAndDownLoadIt() throws InterruptedException {
+        homePage = new HomePage();
+        homePage.signInButton.click();
+        loginPage = new LoginPage();
+        loginPage.getLoginTable().shouldBe(Condition.visible);
+        loginPage.getStudentLogIn();
+        Thread.sleep(1000);
```
Comment (LeonadO555) [inline src/test/java/tests/e2e/TeacherCanUploadCourseMaterialAndStudentCanDownLoadIt.java]: need to replace for waiting for any element

File: src/test/java/tests/e2e/TeacherCanUploadCourseMaterialAndStudentCanDownLoadIt.java [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+        homePage.coursesDropdownMenu.click();
+        homePage.courseListButton.shouldBe(Condition.visible);
+        homePage.courseListButton.click();
+        courseListPage = new CourseListPage();
+        courseListPage.searchBox.shouldBe(Condition.visible);
+        courseListPage.searchCourseByName(facultyName);
+        courseListPage.getSearchedFacultyCourseInTable().shouldBe(Condition.visible);
+        courseListPage.courseCard.click();
+        courseDeatailsPage = new CourseDeatailsPage();
+        Thread.sleep(1000);
```
Comment (LeonadO555) [inline src/test/java/tests/e2e/TeacherCanUploadCourseMaterialAndStudentCanDownLoadIt.java]: need to replace for waiting for any element

File: src/test/java/pages/CourseListPage.java [hunk 6]
```diff
@@ -5,22 +5,31 @@
...
-
-    public List<String>getTitles(){return courseTitles.texts();}
+    public SelenideElement courseListTitle = $x("//h1[normalize-space()='Our courses']");
+    public ElementsCollection courseCarts = $$x("//*[@class=\"list-container\"]");
+    public ElementsCollection courseTitles = $$x("//*[@class=\"list-container\"]//h3");
+    public SelenideElement courseCard = $x("//*[@class=\"list-action-wrapper\"]//a");
+    public SelenideElement searchedFacultyCourseInTable = $x("//*[@class=\"MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 clickable css-isbt42\"]//h3[contains(text(),\"Enjoy QA with Selenide\")]");
+
+    //this method with such locator does not work .
+    public void checkCourseIsVisible(String facultyName){
```
Comment (LeonadO555) [inline src/test/java/pages/CourseListPage.java]: need to do working))

**Review by LeonadO555 — APPROVED** [review]

File: src/test/java/pages/AddNewCoursePage.java (hunk 1 above)
Comment (Kristinochka-bee) [reply src/test/java/pages/AddNewCoursePage.java]: done

File: src/test/java/pages/CourseListPage.java (hunk 6 above)
Comment (Kristinochka-bee) [reply src/test/java/pages/CourseListPage.java]: changed

Comment (Kristinochka-bee) [reply src/test/java/pages/CourseListPage.java]: now is working throught locator

File: src/test/java/tests/e2e/TeacherCanUploadCourseMaterialAndStudentCanDownLoadIt.java (hunk 5 above)
Comment (Kristinochka-bee) [reply src/test/java/tests/e2e/TeacherCanUploadCourseMaterialAndStudentCanDownLoadIt.java]: done

File: src/test/java/tests/e2e/TeacherCanUploadCourseMaterialAndStudentCanDownLoadIt.java (hunk 4 above)
Comment (Kristinochka-bee) [reply src/test/java/tests/e2e/TeacherCanUploadCourseMaterialAndStudentCanDownLoadIt.java]: done

File: src/test/java/tests/e2e/TeacherCanUploadCourseMaterialAndStudentCanDownLoadIt.java (hunk 3 above)
Comment (Kristinochka-bee) [reply src/test/java/tests/e2e/TeacherCanUploadCourseMaterialAndStudentCanDownLoadIt.java]: done

File: src/test/java/pages/CourseDeatailsPage.java (hunk 2 above)
Comment (Kristinochka-bee) [reply src/test/java/pages/CourseDeatailsPage.java]: done

**Review by LeonadO555 — APPROVED** [review]

File: src/test/java/tests/e2e/RegistrationNewUserUITest.java [hunk 7]
```diff
@@ -47,41 +52,41 @@ public void studentRegustrationAndCheckInStudentDirectoryUITest(){
         homePage = new HomePage();
```
Comment (LeonadO555) [inline src/test/java/tests/e2e/RegistrationNewUserUITest.java:52]: studentRegistrationAndCheckInStudentDirectoryUITest

File: src/test/java/tests/e2e/RegistrationNewUserUITest.java [hunk 8]
```diff
@@ -47,41 +52,41 @@ public void studentRegustrationAndCheckInStudentDirectoryUITest(){
...
-        studentDirectoryPage.studentList.contains(studentName);
-        studentDirectoryPage.studentList.contains(UserEmails.STUDENT_NONA_SIRBUL);
+        studentDirectoryPage.header.shouldBe(Condition.visible);
+        studentDirectoryPage.searchBox.sendKeys(studentAnnaBelovaName);
+        studentDirectoryPage.checkStudentTableContainsUserInfo(studentAnnaBelovaName);
+        studentDirectoryPage.checkStudentTableContainsUserInfo(studentAnnaBelovaEmail);
     }
 
     @Test
     public void teacherRegustrationAndCheckInProfessorDirectoryUITest(){
```
Comment (LeonadO555) [inline src/test/java/tests/e2e/RegistrationNewUserUITest.java]: studentRegistrationAndCheckInStudentDirectoryUITest

**Review by LeonadO555 — APPROVED** [review]

File: src/test/java/tests/e2e/RegistrationNewUserUITest.java (hunk 7 above)
Comment (Kristinochka-bee) [reply src/test/java/tests/e2e/RegistrationNewUserUITest.java:52]: fixed

File: src/test/java/tests/e2e/RegistrationNewUserUITest.java (hunk 8 above)
Comment (Kristinochka-bee) [reply src/test/java/tests/e2e/RegistrationNewUserUITest.java]: done

**Pull request merged**